contrib/epee/src/readline_buffer.cpp

```diff
@@ -16,6 +17,8 @@ static std::string last_prompt;
 std::mutex line_mutex, sync_mutex, process_mutex;
 std::condition_variable have_line;
 
+std::vector<std::string> rdln::readline_buffer::completion_commands = {"exit"};
```

Comment:
The public static functions can be invoked from another translation unit which is the source of the problem. The compiler could theoretically resolve the order in some cases with LTO enabled or by marking object files with memory accesses it could "see" from headers. Not aware of Clang or GCC attempting either, based on my searches. The closest I've seen is a GCC attribute marking priority of a global.  I suggested the change to a function scoped static because while it does require/introduce a thread-safe runtime check, switching between the two styles is easy and the overhead is still minimal.

The situation is similar to an [FAQ on isocpp.org](https://isocpp.org/wiki/faq/ctors#static-init-order-on-first-use-members).